### PR TITLE
Fix: diverge global RNG across DDP ranks

### PR DESCRIPTION
**Pull request by danbraunai-goodfire**
## Description

Adds per-rank RNG seeding so that stochastic operations (mask sampling, PGD source init) diverge across DDP ranks, giving more diverse gradient estimates. Model parameters are explicitly broadcast from rank 0 to ensure identical initialization despite divergent RNG.

Changes:
- **`spd/utils/distributed_utils.py`**: Added `seed_per_rank(base_seed)` (uses `base_seed * world_size + rank` for collision-free seeds) and `broadcast_model_params(model)` (broadcasts all params from rank 0).
- **`spd/run_spd.py`**: Call `broadcast_model_params(model)` after `model.to(device)`, then `seed_per_rank(config.seed)` before DDP wrapping and the training loop.
- **`tests/test_seed_per_rank_distributed.py`**: 2 distributed tests verifying RNG divergence across ranks and broadcast param sync.

## Motivation and Context

Previously all DDP ranks shared the same global RNG seed, so stochastic mask sampling produced identical masks on every rank. This wasted the diversity benefit of data parallelism — each rank got different data but identical masks. With per-rank seeding, each rank samples different masks, giving more diverse gradient estimates for the same compute cost.

## How Has This Been Tested?

- 2 new distributed tests (torchrun with 2 ranks on CPU/gloo) verifying: RNG diverges across ranks, `broadcast_model_params` syncs parameters.

## Does this PR introduce a breaking change?

Yes — the global RNG now diverges per rank, which changes the random state during training. Runs with `dp > 1` will produce different results than before (by design). Single-GPU runs are unaffected since `seed * 1 + 0 = seed`.

### CODE REVIEW

File: spd/run_spd.py [hunk 1]
```diff
@@ -162,6 +164,10 @@ def create_pgd_data_iter() -> (
     )
 
     model.to(device)
+    broadcast_model_params(model)
```
[the remark this answers is not in the document]
Comment (danbraunai-goodfire) [reply spd/run_spd.py:167]: Good point. I removed this function entirely in 76b732e4
